### PR TITLE
refactor topological_sort to use Node.get_users()

### PR DESCRIPTION
**Pull request by wenzhe-nrv**
replacing deprecated functions: `Node.get_outputs()`
`get_users()` wraps get_output_size and get_output_inputs and returns a NodeVector

### CODE REVIEW

File: src/ngraph/graph_util.hpp [hunk 1]
```diff
@@ -96,17 +96,11 @@ namespace ngraph
             result_list.push_back(node_map[independent_node]);
             independent_nodes.pop_front();
 
-            for (auto& output : independent_node->get_outputs())
+            for (auto user : independent_node->get_users())
```
Comment (rkimballn1) [inline src/ngraph/graph_util.hpp]: I call auto abuse here. Use `auto&` or better yet `const shared_ptr<Node>&`.  Topological sort is called frequently and creating temporaries can be expensive. Same for line 176

**Comment by wenzhe-nrv** [comment]
@rkimballn1 I also found and fixed a few places in graph_util.cpp that used `auto u: node->get_users()`.
Changed to `for (const auto& u : node->get_users())`

**Comment by wenzhe-nrv** [comment]
@rkimballn1 bob, please review again.

**Comment by wenzhe-nrv** [comment]
This PR is fully reviewed.  Re-requested @louisfeng review by mistake.

**Pull request merged**